### PR TITLE
Search vendor Functionality

### PR DESCRIPTION
**Pull request by czhou578**
Any user will be able to search for a vendor in the top input bar, and as the user types, the results will show under the input box.

Close: https://github.com/bcfoodapp/streetfoodlove/issues/103

### CODE REVIEW

File: app/src/components/UI/Atoms/SearchBox/SearchBox.tsx [hunk 1]
```diff
@@ -1,26 +1,76 @@
...
+    data: SearchProps
+  ) => {
+    if (data.value?.length === 0) {
+      setSearchResult([]);
+      return;
+    }
+
+    if (vendorsList) {
+      for (let i = 0; i < vendorsList.length; i++) {
+        if (data.value === vendorsList[i].Name) {
```
Comment (MakotoE) [inline app/src/components/UI/Atoms/SearchBox/SearchBox.tsx:48]: Shouldn't a vendor be shown in the result if its name contains the string? The search suggestions are only helpful if I don't have to type the entire name.

Comment (czhou578) [reply app/src/components/UI/Atoms/SearchBox/SearchBox.tsx:48]: I can see what you mean, but just a note: the way Selina generated the fake data, the names are like "vendor81", "vendor83", "vendor85" etc. So in order to filter it down, you would have to at least type in the word "vendor".

Comment (MakotoE) [reply app/src/components/UI/Atoms/SearchBox/SearchBox.tsx:48]: I understand that. In real-world usage, names will be very different.

Comment (czhou578) [reply app/src/components/UI/Atoms/SearchBox/SearchBox.tsx:48]: image
image

I added two different names that start with the same two letters and this is how it filters. How is that?

Comment (MakotoE) [reply app/src/components/UI/Atoms/SearchBox/SearchBox.tsx:48]: I think it should match names by substring. You should be able to search between the name and not just by prefix. Lots of names can start with "The" and you shouldn't have to type "The" to search those.

**Review by MakotoE — APPROVED** [review]
`RegExp` should preferably not be used for matching since it throws an error when you type a special character such as parantheses. But because it works for alphanumeric characters, it is fine for now.

**Pull request merged**